### ISSUE TITLE
TagPicker seems to be missing "moreSuggestionsAvailable" property

### ISSUE THREAD

**Issue by gbminnock** [OfficeDev/office-ui-fabric-react#6582]
### Bug Report 
 
I am using the TagPicker component to allow users to select items from a list. I retireve the list items 10 at a time from SharePoint. I set the property searchForMoreText='more...' on the component. When the user clicks on this a second set of results is retrieved and added to the component using the method "onGetMoreResults".

At this point I expect "more..." to still exist so that user can select 3rd set of results. However this is missing.

 
#### Actual behavior:
searchForMoreText is not displayed after second set of results retrieved
 
#### Expected behavior:
 searchForMoreText should be display so that it can be clicked on to get the next set of results.

**Comment by Vitalius1** [comment]
I am looking into [this ](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/pickers/Suggestions/Suggestions.types.ts#L80) prop and trying to see if there is a way to control it as from a surface investigation looks like it's only internally used and not exposed to the surface.

**Comment by gbminnock** [comment]
Hi @Vitalius1 ,
here is a link to the [codepen](https://codepen.io/gbminnock/pen/jeMbaB?editors=0010) I created to demonstrate the issue. Yes if you could see if there is a way to achieve the desired results I would appreciate it. Might it be that the "more" could be only for one time use?

Many Thanks,
Gary

**Comment by Vitalius1** [comment]
@gbminnock one way to control it is add to `pickerSuggestionsProps` a controlled by your app's state boolean value to `moreSuggestionsAvailable` property. In your codepen it should look like this:
```
pickerSuggestionsProps = {
  {
    noResultsFoundText: 'n.a',
    searchForMoreText: 'more...',
    moreSuggestionsAvailable: true
  }
}
```
@joschect From [this](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx#L260-L268) lines it seems that if we add to `pickerSuggestionsProps` the `moreSuggestionsAvailable` prop it will override the whole logic of setting the state for that property. Is this a good pattern to follow or a better approach would be to change the logic of setting the state depending whether the user passed this prop or not?

**Comment by gbminnock** [comment]
@Vitalius1 yes that works in the codepen fine. My current spfx project I cannot use that property and I am using:

- office-ui-fabric-react@5.83.0
- office-ui-fabric-core@9.1.0

Will I have to upgrade the these libraries individually to make use of this property?

thanks again.

**Comment by Vitalius1** [comment]
@gbminnock `office-ui-fabric-core` has nothing to do with it. Let me see if I can find out if in `5.83.0` we had this prop in. Also, I believe you won't be able to update the version to our latest `6.0` release as `spfx` is still on `5.0` (I could be wrong). Plus `6.0` requires `React 16` which again I believe `spfx` is still on `React 15` (I could be wrong again). I will investigate and see if at least the latest of `5.0` which I believe is `5.128.0` has a the prop present.

**Comment by Vitalius1** [comment]
Hmm I verified it in [here](https://codesandbox.io/s/9n227yy8y). It works for me... 
> I cannot use that property
What did you mean by 'cannot'? Is it giving you an error?

**Comment by gbminnock** [comment]
@Vitalius1 when I add the "moreSuggestionAvailable" property I get a typescript error and cannot proceed: 

```typescript
[ts]
Type '{ suggestionsHeaderText: string; noResultsFoundText: string; suggestionsItemClassName: string; suggestionsClassName: string; searchForMoreText: string; resultsMaximumNumber: number; moreSuggestionsAvailable: boolean; }' is not assignable to type 'IBasePickerSuggestionsProps'.
  Object literal may only specify known properties, and 'moreSuggestionsAvailable' does not exist in type 'IBasePickerSuggestionsProps'.
BasePicker.types.d.ts(73, 5): The expected type comes from property 'pickerSuggestionsProps' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<TagPicker> & Readonly<{ children?: ReactNode; }> & Readonly<ITagPickerProps>'
(property) moreSuggestionsAvailable: boolean
```

**Comment by Vitalius1** [comment]
@gbminnock looked once again and it seems this would not work as `moreSuggestionsAvailable` is a property of `ISuggestionsProps` [here](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/pickers/Suggestions/Suggestions.types.ts#L80), but `pickerSuggestionsProps` is of type `IBasePickerSuggestionsProps` in [here](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts#L153) which is not extending from the `ISuggestionsProps`.

@joschect look my comment above. Feels like this was done intentionally as to prevent users to override this state controlled props. Can you advise?

@gbminnock Even if we do a fix in `6.0` branch we could port the fix to the `5.0` branch but you would have to bump to the latest of `5.0`

**Comment by joschect** [comment]
@Vitalius1 Yes it was intentional to prevent it from being overridden. However, I believe that @gbminnock  provides a good use case where there could be multiple pages of results. I think we should consider, in the short term, allowing that property to be overridden, and in the long term provide more support for some kind of paging.

**Comment by Vitalius1** [comment]
@micahgodbolt should we make a fix for now by exposing that prop to users and at the same time file new feature request for pickers to support some kind of paging as @joschect mentioned above?

**Comment by micahgodbolt** [comment]
That sounds good to me. Make sure to mark the new issue with 'needs design' as we'd want to make sure we got some guidance on how the new paging would work

**Comment by Vitalius1** [comment]
@mikewheaton this can be a good candidate for Shield. Make the prop available to the surface API and port this change to `5.0` branch.

**Comment by Matej4386** [comment]
Any news about this? Looking for v5 to be available ...
